examples/applications/plot_digits_denoising.py

```diff
@@ -0,0 +1,136 @@
+"""
+================================
+Image denoising using kernel PCA
+================================
+
+This example shows how to use :class:`~sklearn.decomposition.kernelPCA` to
+denoise images. In short, we take advantage of the approximation function
+learned during `fit` to reconstruct the original image.
+
+We will compare the results with an exact reconstruction using
+:class:`~sklearn.decomposition.PCA`.
+
+We will use USPS digits dataset to reproduce presented in Sect. 4 of [1]_.
+
+.. topic:: References
+
+   .. [1] `Bakır, Gökhan H., Jason Weston, and Bernhard Schölkopf.
+      "Learning to find pre-images."
+      Advances in neural information processing systems 16 (2004): 449-456.
+      <https://papers.nips.cc/paper/2003/file/ac1ad983e08ad3304a97e147f522747e-Paper.pdf>`_
+"""
+
+print(__doc__)
+
+# Authors: Guillaume Lemaitre <guillaume.lemaitre@inria.fr>
+# Licence: BSD 3 clause
+
+# %%
+# Load the dataset via OpenML
+# ---------------------------
+#
+# The USPS digits datasets in available in OpenML. We use
+# :func:`~sklearn.datasets.fetch_openml` to get this dataset. In addition, we
+# normalize the dataset such that all pixel values are in the range (0, 1).
+import numpy as np
+from sklearn.datasets import fetch_openml
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.model_selection import train_test_split
+
+X, y = fetch_openml(data_id=41082, as_frame=False, return_X_y=True)
+X = MinMaxScaler().fit_transform(X)
+
+# %%
+# The idea will be to learn a PCA basis (with and without a kernel) on denoised
```

Comment:
The following would be more accurate:

```suggestion
# The idea will be to learn a PCA basis (with and without a kernel) on noise-free
```

However, isn't this cheating? In "real" life use cases, access to noise free images is not possible. Have you tried to fit the PCA and kPCA model on noisy images instead?

Reply:
Then, we don't reproduce the experiment of the paper. I can try to see what we get

Reply:
If that does not work, maybe just add a note to say that the setup is a bit artificial but the goal of this example is to try to reproduce the experiment of the paper.

Reply:
So I switched the setting to be more realistic and this is working as before. I adapted the narration.